### PR TITLE
feat: [DHIS2-12514] Add new option in enrollment widget

### PR DESCRIPTION
**Pull request by jasminenguyennn**
### Feature 
[DHIS2-12514](https://jira.dhis2.org/browse/DHIS2-12514): Implement "Add new" for Completed enrollments in Enrollment Widget

The Add new button will navigate to  register page (i.e: `new?orgUnitId=DiszpKrYNg8&programId=ur1Edk5Oe2n&teiId=nj37aOvo3Qq`)
as the widget is self-contained, prefill the data is not yet possible, this should be done after [DHIS2-12141](https://jira.dhis2.org/browse/DHIS2-12141) is done. Right now I redirect the page to the correct teiId

### if only enroll once 
Screenshot 2022-02-22 at 07 45 43

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetEnrollment/Actions/Actions.container.js [hunk 1]
```diff
@@ -34,12 +37,19 @@ export const Actions = ({ enrollment = {}, refetch, onDelete }: Props) => {
             onComplete: onDelete,
         },
     );
+    const onHandleAddNew = () => {
+        const { programId, orgUnitId, teiId } = deriveURLParamsFromLocation();
+        history.push(`/new?${buildUrlQueryString({ programId, orgUnitId, teiId })}`);
```
[the remark this answers is not in the document]
Comment (jasminenguyennn) [reply src/core_modules/capture-core/components/WidgetEnrollment/Actions/Actions.container.js]: @simonadomnisoru yes that makes sense, I moved the onAddNew to the page level

**Review by simonadomnisoru — APPROVED** [review]

File: src/core_modules/capture-core/components/WidgetEnrollment/Actions/AddNew/AddNew.component.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    dense
+    dataTest="widget-enrollment-actions-add-new"
+    onClick={onClick}
+    icon={<IconAdd16 />}
+    label={i18n.t('Add new')}
+    {...passOnProps}
+/>);
+
+export const AddNew = ({ enrollment, onlyEnrollOnce, tetId, onAddNew }: Props) => {
+    const isCompleted = enrollment.status === plainStatus.COMPLETED;
```
[the remark this answers is not in the document]
Comment (jasminenguyennn) [reply src/core_modules/capture-core/components/WidgetEnrollment/Actions/AddNew/AddNew.component.js]: thanks @JoakimSM for clarifying this, the logic is: if there is no active enrollment in the tei, there should be option to `Add new`. 
I adjust the code to reflect that 👍

File: src/core_modules/capture-core/components/WidgetEnrollment/Actions/AddNew/AddNew.component.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    icon={<IconAdd16 />}
+    label={i18n.t('Add new')}
+    {...passOnProps}
+/>);
+
+export const AddNew = ({ canAddNew, onlyEnrollOnce, tetId, onAddNew }: Props) => {
+    if (!canAddNew) {
+        return null;
+    }
+    const tetName = tetId ? getTrackedEntityTypeThrowIfNotFound(tetId)?.name : '';
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/WidgetEnrollment/Actions/AddNew/AddNew.component.js:21]: To keep this self-contained, we should get the TET name in src/core_modules/capture-core/components/WidgetEnrollment/hooks/useProgram.js and use that one (add `trackedEntityType[displayName]` to the fields)

**Review by JoakimSM — COMMENTED** [review]
One change request, looks good apart from that 👍

**Review by JoakimSM — APPROVED** [review]
Looks good 👍

**Pull request merged**